pkg/catalog/catalog.go

```diff
@@ -1748,67 +1753,213 @@ func (c *Catalog) FindMergeBase(ctx context.Context, repositoryID string, destin
+func (c *Catalog) DumpRefsStatus(ctx context.Context, repositoryID string, id string) (*RefsDumpStatus, error) {
 	repository, err := c.getRepository(ctx, repositoryID)
 	if err != nil {
-		return "", err
+		return nil, err
+	}
+	if !isTaskID(DumpRefsTaskIDPrefix, id) {
+		return nil, graveler.ErrNotFound
 	}
 
-	metaRangeID, err := c.Store.DumpBranches(ctx, repository)
+	var taskStatus RefsDumpStatus
+	err = getTaskStatus(ctx, c.KVStore, repository, id, &taskStatus)
 	if err != nil {
-		return "", err
+		return nil, err
 	}
-	return string(*metaRangeID), nil
+	return &taskStatus, nil
 }
 
-func (c *Catalog) DumpTags(ctx context.Context, repositoryID string) (string, error) {
+func (c *Catalog) RestoreRefsSubmit(ctx context.Context, repositoryID string, info *RefsDumpInfo) (string, error) {
 	repository, err := c.getRepository(ctx, repositoryID)
 	if err != nil {
 		return "", err
 	}
-
-	metaRangeID, err := c.Store.DumpTags(ctx, repository)
-	if err != nil {
+	// create refs restore task and update initial status
+	taskID := newTaskID(RestoreRefsTaskIDPrefix)
+	taskStatus := &RefsDumpStatus{
+		Task: &Task{
+			Id:        taskID,
+			Completed: false,
+			UpdatedAt: timestamppb.Now(),
+		},
+		Info: &RefsDumpInfo{},
+	}
+	if err := updateTaskStatus(ctx, c.KVStore, repository, taskID, taskStatus); err != nil {
 		return "", err
 	}
-	return string(*metaRangeID), nil
-}
 
-func (c *Catalog) LoadCommits(ctx context.Context, repositoryID, commitsMetaRangeID string) error {
-	repository, err := c.getRepository(ctx, repositoryID)
-	if err != nil {
-		return err
+	c.workPool.Submit(func() {
+		c.restoreRefsTask(ctx, repository, taskID, taskStatus, info)
+	})
+	return taskID, nil
+}
+
+func (c *Catalog) restoreRefsTask(ctx context.Context, repository *graveler.RepositoryRecord, taskID string, taskStatus *RefsDumpStatus, info *RefsDumpInfo) {
+	steps := []struct {
+		name string
+		fn   func() error
+	}{
+		{
+			name: "load commits",
+			fn: func() error {
+				return c.Store.LoadCommits(ctx, repository, graveler.MetaRangeID(info.CommitsMetarangeId))
+			},
+		},
+		{
+			name: "load branches",
+			fn: func() error {
+				return c.Store.LoadBranches(ctx, repository, graveler.MetaRangeID(info.BranchesMetarangeId))
+			},
+		},
+		{
+			name: "load tags",
+			fn: func() error {
+				return c.Store.LoadTags(ctx, repository, graveler.MetaRangeID(info.TagsMetarangeId))
+			},
+		},
+	}
+
+	// run each step and update step status
+	for stepIdx, step := range steps {
+		err := step.fn()
+		if err != nil {
+			c.log(ctx).
+				WithError(err).
+				WithFields(logging.Fields{"task_id": taskID, "repository": repository.RepositoryID}).
+				Error("failed to %s", step.name)
+			taskStatus.Task.Error = err.Error()
+			taskStatus.Task.Completed = true
+		} else if stepIdx == len(steps)-1 {
+			taskStatus.Task.Completed = true
+		}
+		taskStatus.Task.UpdatedAt = timestamppb.Now()
```

Comment:
Move all assignments of "UpdatedAt" into the updateTaskStatus function?

Reply:
Didn't want the function that update the kv to update my message - let me know if you think we should update the function purpose.